### PR TITLE
Ignore public methods with a single non-class arguments during auto r…

### PR DESCRIPTION
**Pull request by weph**
Here's the PR, @ruudk

### CODE REVIEW

File: Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php [hunk 1]
```diff
@@ -54,16 +53,11 @@ public function process(ContainerBuilder $container)
 
                     $parameters = $method->getParameters();
 
-                    // if no param or optional param, skip
-                    if (count($parameters) !== 1 || $parameters[0]->isOptional()) {
+                    // if no param, optional param or non-class param, skip
+                    if (count($parameters) !== 1 || $parameters[0]->isOptional() || is_null($parameters[0]->getClass())) {
```
Comment (ruudk) [inline Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php]:
```suggestion
                    if (count($parameters) !== 1 || $parameters[0]->isOptional() || $parameters[0]->getClass() === null) {
```

File: Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php [hunk 2]
```diff
@@ -54,16 +53,11 @@ public function process(ContainerBuilder $container)
 
                     $parameters = $method->getParameters();
 
-                    // if no param or optional param, skip
-                    if (count($parameters) !== 1 || $parameters[0]->isOptional()) {
+                    // if no param, optional param or non-class param, skip
+                    if (count($parameters) !== 1 || $parameters[0]->isOptional() || $parameters[0]->getClass() === null) {
```
Comment (AlessandroMinoccheri) [inline Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php]: Could it be useful to create a method named correctly  for this if so we can remove the comment, make the code more clean and readable? What do you think?

Comment (weph) [reply Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php]: Was thinking the same. But I'd rather extract all the logic the determines whether a given `ReflectionMethod` is a valid handler. For my taste the whole process method is too long and nested but I didn't want to dig it all over for this small change.

Comment (AlessandroMinoccheri) [reply Bundle/SymfonyBridge/src/DependencyInjection/Compiler/AutoRegister.php]: Ok thanks for the explanation

**Comment by ruudk** [comment]
Thanks @weph

**Pull request merged**